sdks/java/core/src/main/java/org/apache/beam/sdk/Pipeline.java

```diff
@@ -380,6 +524,32 @@ public void traverseTopologically(PipelineVisitor visitor) {
   }
 
   /**
+   * retrieve {@code Pipeline}, and print out data flow in visualized format.
+   *
+   * <p>Give a pipeline like
+   * <pre>{@code
+   * Pipeline pipeline = Pipeline.create(...);
+   * PCollection<?> stream = pipeline.apply("source",IO.<?>read())
+   *   .apply("process_step", ParDo.of(...));
+   * stream.apply("persistent", IO.<?>write());
```

Comment:
This is an example which joins two sources, and applied two filters separately on the join result.
```
[end] filte_one/ParMultiDo(Anonymous)
    ParDo(Anonymous)/ParMultiDo(Anonymous)
        CoGroupByKey/ConstructCoGbkResultFn/ParMultiDo(ConstructCoGbkResult)
            CoGroupByKey/GroupByKey
                CoGroupByKey/Flatten.PCollections
                    CoGroupByKey/MakeUnionTable0/ParMultiDo(ConstructUnionTable)
                        Window.Into()/Window.Assign
                            ParDo(ListingItemPartExtract)/ParMultiDo(ListingItemPartExtract)
                                Values/Values/Map/ParMultiDo(Anonymous)
                                    item_topic/Remove Kafka Metadata/ParMultiDo(Anonymous)
                                        item_topic/Read(UnboundedKafkaSource)
                    CoGroupByKey/MakeUnionTable1/ParMultiDo(ConstructUnionTable)
                        Window.Into()2/Window.Assign
                            ParDo(ListingVariationPartExtract)/ParMultiDo(ListingVariationPartExtract)
                                Values2/Values/Map/ParMultiDo(Anonymous)
                                    variation_topic/Remove Kafka Metadata/ParMultiDo(Anonymous)
                                        variation_topic/Read(UnboundedKafkaSource)
[end] filter_two/ParMultiDo(Anonymous)
    ParDo(Anonymous)/ParMultiDo(Anonymous)
        CoGroupByKey/ConstructCoGbkResultFn/ParMultiDo(ConstructCoGbkResult)
            CoGroupByKey/GroupByKey
                CoGroupByKey/Flatten.PCollections
                    CoGroupByKey/MakeUnionTable0/ParMultiDo(ConstructUnionTable)
                        Window.Into()/Window.Assign
                            ParDo(ListingItemPartExtract)/ParMultiDo(ListingItemPartExtract)
                                Values/Values/Map/ParMultiDo(Anonymous)
                                    item_topic/Remove Kafka Metadata/ParMultiDo(Anonymous)
                                        item_topic/Read(UnboundedKafkaSource)
                    CoGroupByKey/MakeUnionTable1/ParMultiDo(ConstructUnionTable)
                        Window.Into()2/Window.Assign
                            ParDo(ListingVariationPartExtract)/ParMultiDo(ListingVariationPartExtract)
                                Values2/Values/Map/ParMultiDo(Anonymous)
                                    variation_topic/Remove Kafka Metadata/ParMultiDo(Anonymous)
                                        variation_topic/Read(UnboundedKafkaSource)
```